llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -203,9 +205,9 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
 
 private:
   ResourceScoreboard<FuncUnitWrapper> Scoreboard;
-  std::map<MachineInstr *, unsigned> SelectedAltOpcodes;
   const AIEBaseInstrInfo *TII;
   const InstrItineraryData *ItinData;
+  AIEAlternateDescriptors *SelectedAltDescs;
```

Comment:
I think we always expect this to be non-null. Could you make that a reference? Same for `AIEAlternateDescriptors *getSelectedAltDescs()`